sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -0,0 +1,594 @@
+    ///
+    ///   The <see cref="EventHubBufferedProducerClient"/> is safe to cache and use as a singleton for the lifetime of an
+    ///   application. This is the recommended approach, since the client is responsible for efficient network,
+    ///   CPU, and memory use. Calling <see cref="CloseAsync(bool, CancellationToken)"/> or <see cref="DisposeAsync"/>
+    ///   is required so that resources can be cleaned up after use.
+    /// </remarks>
+    ///
+    /// <seealso cref="EventHubProducerClient" />
+    ///
+    internal class EventHubBufferedProducerClient : IAsyncDisposable
+    {
+        private static EventHubBufferedProducerClientOptions DefaultOptions { get; } = new EventHubBufferedProducerClientOptions
+        {
+            RetryOptions = new EventHubsRetryOptions { MaximumRetries = 15, TryTimeout = TimeSpan.FromMinutes(3) }
+        };
+
+        /// <summary>
+        ///   The fully qualified Event Hubs namespace that this producer is currently associated with, which will likely be similar
+        ///   to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public string FullyQualifiedNamespace => _producer.FullyQualifiedNamespace;
+
+        /// <summary>
+        ///   The name of the Event Hub that this producer is connected to, specific to the Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName => _producer.EventHubName;
+
+        /// <summary>
+        ///   A unique name to identify the buffered producer.
+        /// </summary>
+        ///
+        public string Identifier => _producer.Identifier;
+
+        /// <summary>
+        ///   The total number of events that are currently buffered and waiting to be published, across all partitions.
+        /// </summary>
+        ///
+        public virtual int TotalBufferedEventCount { get; private set; }
+
+        /// <summary>
+        ///   <c>true</c> if the producer has been closed <c>false</c> otherwise.
+        /// </summary>
+        ///
+        public bool IsClosed => _isClosed;
+
+        /// <summary>
+        ///    Invoked after each batch of events has been successfully published to the Event Hub, this
+        ///    handler is optional and is intended to provide notifications for interested listeners.
+        /// </summary>
+        ///
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "Guidance does not apply; this is an event.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0003:DO make service methods virtual.", Justification = "This member follows the standard .NET event pattern; override via the associated On<<EVENT>> method.")]
+        public event Func<SendEventBatchSuccessEventArgs, Task> SendEventBatchSuccessAsync
```

Comment:
Can we move both event declarations after the class members but before the properties?